components/bitcoin-topics/markdown-ui/bitcoin-history.tsx

```diff
@@ -0,0 +1,587 @@
+"use client"
+
+import React, { useState, useEffect } from "react"
+import { motion, useAnimation } from "framer-motion"
+import { useInView } from "react-intersection-observer"
+import Image from "next/image"
+import { ExternalLinkIcon } from "lucide-react"
+
+interface HistoryEvent {
+    id: string
+    type: string
+    date: string
+    title: string
+    paragraphs: string[]
+    links: { label: string; link: string }[]
+    media?: {
+        type: "image" | "video" | "svg" | "gif"
+        src: string
+        alt: string
+    }
+}
+
+interface EventTypeBadgeProps {
+    type: string
+}
+
+const EventTypeBadge: React.FC<EventTypeBadgeProps> = ({ type }) => {
+    const getBadgeColor = (type: string) => {
+        switch (type.toLowerCase()) {
+            case "post":
+                return "bg-orange-400 dark:bg-orange-600"
+            case "release":
+                return "bg-vscode-success-light dark:bg-vscode-success-dark"
+            case "bug":
+                return "bg-vscode-error-light dark:bg-vscode-error-dark"
+            case "other":
+                return "bg-gray-400 dark:bg-gray-600"
+            default:
+                return "bg-vscode-text-light dark:bg-vscode-text-dark"
+        }
+    }
+
+    return (
+        <span
+            className={`inline-block px-2 py-1 text-xs font-semibold text-white rounded-full ${getBadgeColor(type)}`}
+        >
+            {type.toUpperCase()}
+        </span>
+    )
+}
+
+const bitcoinHistory: HistoryEvent[] = [
```

Comment:
What are your thoughts regarding this data being available on `bitcoin-topics` repo instead.

currently suggest-edits action navigates the user to `https://github.com/jrakibi/bitcoin-topics/edit/main/topics/bitcoin-history.mdx` (I manually checked the feature/project branch for the file). And it has only the metadata and `<BitcoinHistory />` component.

My suggestion is the data should be passed from that mdx file e.g
```mdx
---
title: "Bitcoin History"
date: 2024-07-05T15:32:14Z
...
---

<BitcoinHistory data={[
  {
    id: "2007-post-satoshi-starts",
    type: "post",
    ...
  }
]}/>
```

What do you think?

Reply:
that's a great idea for separation of concerns. It allows us to update the data without affecting the `bitcoin-dev-project` repo